src/query/src/error.rs

```diff
@@ -316,9 +316,13 @@ impl ErrorExt for Error {
             ParseSql { source, .. } => source.status_code(),
             CreateRecordBatch { source, .. } => source.status_code(),
             QueryExecution { source, .. } | QueryPlan { source, .. } => source.status_code(),
-            DataFusion { .. } | MissingTimestampColumn { .. } | RoutePartition { .. } => {
-                StatusCode::Internal
-            }
+            DataFusion { error, .. } => match error {
+                DataFusionError::Internal(_) => StatusCode::Internal,
```

Comment:
Should we map datafusion's internal error to our internal code? I think `EngineExecuteQuery` is fine.